make/_shared/tools/00_mod.mk

```diff
@@ -381,10 +381,10 @@ $(call for_each_kv,go_dependency,$(go_dependencies))
 # File downloads #
 ##################
 
-go_linux_amd64_SHA256SUM=9379441ea310de000f33a4dc767bd966e72ab2826270e038e78b2c53c2e7802d
-go_linux_arm64_SHA256SUM=561c780e8f4a8955d32bf72e46af0b5ee5e0debe1e4633df9a03781878219202
-go_darwin_amd64_SHA256SUM=782da50ce8ec5e98fac2cd3cdc6a1d7130d093294fc310038f651444232a3fb0
-go_darwin_arm64_SHA256SUM=5cae2450a1708aeb0333237a155640d5562abaf195defebc4306054565536221
+go_linux_amd64_SHA256SUM=45b87381172a58d62c977f27c4683c8681ef36580abecd14fd124d24ca306d3f
+go_linux_arm64_SHA256SUM=9d6d938422724a954832d6f806d397cf85ccfde8c581c201673e50e634fdc992
+go_darwin_amd64_SHA256SUM=4a0f0a5eb539013c1f4d989e0864aed45973c0a9d4b655ff9fd56013e74c1303
+go_darwin_arm64_SHA256SUM=d4f53dcaecd67d9d2926eab7c3d674030111c2491e68025848f6839e04a4d3d1
```

Comment:
I updated these by running `make vendor-go`, noting the sha mismatch and then pasting the actual sha into this file.

```
make vendor-go HOST_ARCH=amd64HOST_OS=linux
make vendor-go HOST_ARCH=arm64 HOST_OS=linux
make vendor-go HOST_ARCH=amd64 HOST_OS=darwin
make vendor-go HOST_ARCH=arm64 HOST_OS=darwin
```

E.g.

```
$ make vendor-go
invalid checksum for "/home/richard/projects/cert-manager/cert-manager/_bin/downloaded/tools/go@1.23.8_linux_amd64.tar.gz.tmp": wanted "9379441ea310de000f33a4dc767bd966e72ab2826270e038e78b2c53c2e7802d" but got "45b87381172a58d62c977f27c4683c8681ef36580abecd14fd124d24ca306d3f"
make: *** [make/_shared/tools/00_mod.mk:391: /home/richard/projects/cert-manager/cert-manager/_bin/downloaded/tools/go@1.23.8_linux_amd64.tar.gz] Error 1
```